테이블 별 데이터 수 확인

In [4]:
import pymysql

# 데이터베이스 연결 설정
connection = pymysql.connect(
    host=,
    port=,
    user=,
    password=,
    database=,
    charset=
)

In [5]:
tables = ['tb_address', 'tb_parking', 'tb_traffic']

try:
    with connection.cursor() as cursor:
        for table in tables:
            count_query = f"SELECT COUNT(*) FROM {table};"
            cursor.execute(count_query)
            total_rows = cursor.fetchone()[0]
            print(f"테이블 '{table}의 전체 행 개수: {total_rows}")
            
except Exception as e:
    print(f"오류 {e}")
    
finally:
    connection.close()

테이블 'tb_address의 전체 행 개수: 528278
테이블 'tb_parking의 전체 행 개수: 94434
테이블 'tb_traffic의 전체 행 개수: 15849288


테이블 별 데이터 확인 (상위20개)

In [6]:
import pymysql
import pandas as pd

# 데이터베이스 연결 설정
connection = pymysql.connect(
    host=,
    port=,
    user=,
    password=,
    database=,
    charset=
)

# 테이블 목록
tables = ['tb_address', 'tb_parking', 'tb_traffic']

try:
    with connection.cursor() as cursor:
        for table in tables:
            print(f"테이블 '{table}' 상위 20개")
            
            query = f"SELECT * FROM {table} LIMIT 20;"
            cursor.execute(query)
            rows = cursor.fetchall()
            
            columns_query = f"SHOW COLUMNS FROM {table};"
            cursor.execute(columns_query)
            columns = [column[0] for column in cursor.fetchall()]
            
            df = pd.DataFrame(rows, columns=columns)
            print(df)
except Exception as e:
    print(f"오류{e}")

finally:
    connection.close()

테이블 'tb_address' 상위 20개
                     address zipcode building_name               lat  \
0     서울특별시 강남구 역삼동 강남대로 298    6253       KB라이프타워  37.4900614477694   
1     서울특별시 강남구 역삼동 강남대로 302    6253          동희빌딩  37.4903138653759   
2   서울특별시 강남구 역삼동 강남대로 302-2    6253          None  37.4903503986844   
3     서울특별시 강남구 역삼동 강남대로 308    6253        랜드마크타워  37.4906711637601   
4     서울특별시 강남구 역삼동 강남대로 310    6253     유니온센터오피스텔  37.4909831820412   
5     서울특별시 강남구 역삼동 강남대로 314    6253          서우빌딩  37.4913189579708   
6     서울특별시 강남구 역삼동 강남대로 318    6253      TOWER837  37.4915693683974   
7   서울특별시 강남구 역삼동 강남대로 318-2    6253         가로판매대  37.4915773626553   
8     서울특별시 강남구 역삼동 강남대로 320    6252          황화빌딩  37.4918533105983   
9     서울특별시 강남구 역삼동 강남대로 324    6252      역삼디오슈페리움  37.4921449015471   
10  서울특별시 강남구 역삼동 강남대로 324-2    6252          None  37.4919158288000   
11  서울특별시 강남구 역삼동 강남대로 326-2    6252          None  37.4921613780665   
12    서울특별시 강남구 역삼동 강남대로 328    6252    

판다스 df로 로드

In [7]:
import pandas as pd
import pymysql

connection = pymysql.connect(
    host=,
    port=,
    user=,
    password=,
    database=,
    charset=
)

# 각 테이블을 개별적으로 df로 불러올 변수
df_address, df_parking, df_traffic = None, None, None

try:
    df_address = pd.read_sql("SELECT * FROM tb_address;", connection)
    df_parking = pd.read_sql("SELECT * FROM tb_parking;", connection)
    df_traffic = pd.read_sql("SELECT * FROM tb_traffic;", connection)
    
    print("tb_address")
    print(df_address.head())
    
    print("tb_parking")
    print(df_parking.head())
    
    print("tb_traffic")
    print(df_traffic.head())
    
except Exception as e:
    print(f"오류{e}")
    
finally:
    connection.close()

C:\Users\Public\Documents\ESTsoft\CreatorTemp\ipykernel_10072\1906281247.py:17: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  df_address = pd.read_sql("SELECT * FROM tb_address;", connection)


오류Execution failed on sql: SELECT * FROM tb_address;
(2013, 'Lost connection to MySQL server during query ([WinError 10054] 현재 연결은 원격 호스트에 의해 강제로 끊겼습니다)')
unable to rollback


Output is truncated. View as a scrollable element or open in a text editor. Adjust cell output settings...
C:\Users\Public\Documents\ESTsoft\CreatorTemp\ipykernel_10072\1906281247.py:17: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  df_address = pd.read_sql("SELECT * FROM tb_address;", connection)
오류Execution failed on sql: SELECT * FROM tb_address;
(2013, 'Lost connection to MySQL server during query ([WinError 10054] 현재 연결은 원격 호스트에 의해 강제로 끊겼습니다)')
unable to rollback

위 오류 발생으로 인해 sqlalchemy 모듈을 사용

In [11]:
%pip install sqlalchemy

Note: you may need to restart the kernel to use updated packages.


In [10]:
import pandas as pd
from sqlalchemy import create_engine

# SQLAlchemy 엔진 생성
connection_string = 'mysql+pymysql://root:?charset=utf8'
engine = create_engine(connection_string)

try:
    df_address = pd.read_sql("SELECT * FROM tb_address;", engine)
    df_parking = pd.read_sql("SELECT * FROM tb_parking;", engine)
    df_traffic = pd.read_sql("SELECT * FROM tb_traffic;", engine)

    print("tb_address")
    print(df_address.head())

    print("tb_parking")
    print(df_parking.head())

    print("tb_traffic")
    print(df_traffic.head())

except Exception as e:
    print(f"오류: {e}")

finally:
    # 엔진 종료
    engine.dispose()


tb_address
                    address zipcode building_name        lat         lng
0    서울특별시 강남구 역삼동 강남대로 298    6253       KB라이프타워  37.490061  127.032168
1    서울특별시 강남구 역삼동 강남대로 302    6253          동희빌딩  37.490314  127.031666
2  서울특별시 강남구 역삼동 강남대로 302-2    6253          None  37.490350  127.031507
3    서울특별시 강남구 역삼동 강남대로 308    6253        랜드마크타워  37.490671  127.031820
4    서울특별시 강남구 역삼동 강남대로 310    6253     유니온센터오피스텔  37.490983  127.031486
tb_parking
       park_name     park_address        lat         lng park_type  capacity  \
0  도봉산역 공영주차장(시)   도봉구 도봉동 288-19  37.689721  127.044839        NW       358   
1   도봉산 공영주차장(시)   도봉구 도봉동 282-26  37.687644  127.040660        NW       142   
2        현대성우아파트   도봉구 도봉동 642-0   37.676229  127.047725        BS       228   
3  당고개위 공영주차장(시)  노원구 상계동 111-568  37.670703  127.080418        NS        23   
4     도봉구청 부설주차장   도봉구 방학동 720-0   37.668716  127.047115        BP       310   

  que_status     park_phone   rates  cur_parking  ... day_m

불러온 데이터 확인

In [12]:
df_address

address zipcode building_name        lat         lng
0          서울특별시 강남구 역삼동 강남대로 298    6253       KB라이프타워  37.490061  127.032168
1          서울특별시 강남구 역삼동 강남대로 302    6253          동희빌딩  37.490314  127.031666
2        서울특별시 강남구 역삼동 강남대로 302-2    6253          None  37.490350  127.031507
3          서울특별시 강남구 역삼동 강남대로 308    6253        랜드마크타워  37.490671  127.031820
4          서울특별시 강남구 역삼동 강남대로 310    6253     유니온센터오피스텔  37.490983  127.031486
...                           ...     ...           ...        ...         ...
528273    서울특별시 송파구 잠실동 도곡로64길 11    5574          None  37.505194  127.079414
528274    서울특별시 송파구 잠실동 도곡로64길 12    5573          None  37.505001  127.079336
528275    서울특별시 송파구 잠실동 도곡로64길 13    5574          None  37.505155  127.079539
528276    서울특별시 송파구 잠실동 도곡로64길 14    5573         욱빈하우스  37.504960  127.079453
528277  서울특별시 송파구 잠실동 도곡로64길 14-1    5573          태영주택  37.504922  127.079571

[528278 rows x 5 columns]

In [13]:
df_parking

park_name     park_address        lat         lng park_type  \
0             도봉산역 공영주차장(시)   도봉구 도봉동 288-19  37.689721  127.044839        NW   
1              도봉산 공영주차장(시)   도봉구 도봉동 282-26  37.687644  127.040660        NW   
2                   현대성우아파트   도봉구 도봉동 642-0   37.676229  127.047725        BS   
3             당고개위 공영주차장(시)  노원구 상계동 111-568  37.670703  127.080418        NS   
4                도봉구청 부설주차장   도봉구 방학동 720-0   37.668716  127.047115        BP   
...                     ...              ...        ...         ...       ...   
94429               팜모드 주차장   강남구 세곡동 551-0   37.467241  127.099975        BS   
94430  이마트 양재점[양재시민의숲역 2호점]   서초구 양재동 215-0   37.463317  127.036871        BS   
94431                서초케이타운   서초구 신원동 559-0   37.457159  127.058727        BS   
94432               내곡드림시티1   서초구 신원동 205-1   37.447350  127.056302        BS   
94433               내곡프라임시티   서초구 신원동 609-0   37.447920  127.055722        BS   

       capacity que_status     park_phone   rates  cur_parking  ...  \
0           358        20분   02)2290-6047   170.0          189  ...   
1           142        20분  010-4927-3981   170.0          151  ...   
2           228        20분   02-3491-8954  1000.0          228  ...   
3            23        20분  010-4927-3981   260.0            6  ...   
4           310        20분   02-2091-2143     0.0          295  ...   
...         ...        ...            ...     ...          ...  ...   
94429        45        20분                 1000.0           45  ...   
94430       649        20분                 1000.0          235  ...   
94431        87        20분                 1500.0           87  ...   
94432        44        20분                 1000.0           44  ...   
94433        38        20분                 1500.0           38  ...   

      day_maximum  time_rate  add_rates  fulltime_monthly  weekday_begin_time  \
0         12200.0        5.0      170.0           91000.0                   0   
1         12200.0        5.0      170.0          126000.0                 800   
2         10000.0       30.0      500.0               NaN                   0   
3             0.0        5.0      260.0               0.0                 900   
4             0.0       60.0      250.0               0.0                 900   
...           ...        ...        ...               ...                 ...   
94429     30000.0       20.0      500.0               0.0                   0   
94430     20000.0       10.0     1000.0          100000.0                1000   
94431     50000.0       30.0      500.0               0.0                   0   
94432     20000.0       30.0      500.0               0.0                   0   
94433     50000.0       30.0      500.0               0.0                   0   

       weekday_end_time  weekend_begin_time  weekend_end_time  \
0                  2400                   0              2400   
1                  2200                 800              2200   
2                  2400                   0              2400   
3                  2200                 900              2200   
4                  1800                 900              1800   
...                 ...                 ...               ...   
94429              2400                   0              2400   
94430              2300                1000              2300   
94431              2400                   0              2400   
94432              2400                   0              2400   
94433              2400                   0              2400   

       holiday_begin_time  holiday_end_time  
0                       0              2400  
1                     800              2200  
2                       0              2400  
3                     900              2200  
4                       0              2400  
...                   ...               ...  
94429                   0              2400  
94430                1000              2300  
94431                   

In [14]:
df_traffic

traffic_id   roadName roadDrcType linkNo      linkId  startNodeId  \
0              35321         종로        None   None  1000000301   1000008600   
1              35322         종로        None   None  1000000302   1000008900   
2              35323         종로        None   None  1000000303   1000007700   
3              35324         종로        None   None  1000000401   1000008900   
4              35325         종로        None   None  1000000402   1000009500   
...              ...        ...         ...    ...         ...          ...   
15849283    15884604        벌말로        None   None  2420271300   2420014901   
15849284    15884605        평천로        None   None  2420271401   2420087801   
15849285    15884606        평천로        None   None  2420271501   2420006500   
15849286    15884607  조마루로386번길        None   None  2420271601   2420033900   
15849287    15884608  조마루로386번길        None   None  2420271701   2420087901   

           endNodeId  speed  travelTime     createdDate  
0         1000008900   11.5       47.24  20241001184000  
1         1000007700   10.0       37.76  20241001184000  
2         1000008700    7.5       32.40  20241001184000  
3         1000008600    7.0       77.61  20241001184000  
4         1000008900    9.0       41.96  20241001184000  
...              ...    ...         ...             ...  
15849283  1150094100   59.0       53.54  20241006155500  
15849284  2420006500   65.0        1.83  20241006155500  
15849285  2420087801   32.0        3.70  20241006155500  
15849286  2420087901   19.0       12.32  20241006155500  
15849287  2420033900   24.0        9.68  20241006155500  

[15849288 rows x 10 columns]

In [15]:
df_address.to_csv('sample_df_address.csv', index=False)
df_parking.to_csv('sample_df_parking.csv', index=False)
df_traffic.to_csv('sample_df_traffic.csv', index=False)